src/ORM/FieldType/DBDatetime.php

```diff
@@ -228,6 +228,38 @@ public function getFormatter($dateLength = IntlDateFormatter::MEDIUM, $timeLengt
         return new IntlDateFormatter(i18n::get_locale(), $dateLength, $timeLength);
```

Comment:
Probably should replace `i18n::get_locale()` with `DBDate::ISO_LOCALE`, here and `DBDate::getCustomFormatter()`

Reply:
The getFormatter() relies on the current locale otherwise it won't localise dates for the frontend.

Reply:
I see, but I have two points:
*calling $date->Format() for other purposes -below use case- will cause the same problem
`$date = DBDatetime::now();
$anotherJd = GregorianToJD($date->Format('M'), $date->Format('d'), $date->Format('Y'));`

* I think there should be a way to set the numerals system away from the locale, so in my case I want Arabic Numerals with Arabic locale, some may want Hindi Numerals with Arabic locale?

Reply:
Yes I thought that would be the case.

For those specific methods we should allow a locale to be specified as well. That way you can use ->Format("someformat") (uses current locale) or ->Format("someformate", "en_NZ") (uses format in nz locale).

For methods that internally use Format() to output numeric values (e.g. day of month) we should similarly use the hard-coded ISO_LOCALE.

Reply:
@hudhaifas you can now pass $locale (e.g. DBDate::ISO_LOCALE) as a second arg to ->Format(), so your use case is now covered. :)

Tests have been added to prove that the issues are fixed, both localising to hindi numerals, as well as internal storage when arabic is set as the parent locale.